build.rs

```diff
@@ -198,8 +197,11 @@ fn load_cross_compile_info() -> Result<(InterpreterConfig, HashMap<String, Strin
 /// Examine python's compile flags to pass to cfg by launching
 /// the interpreter and printing variables of interest from
 /// sysconfig.get_config_vars.
-#[cfg(not(target_os = "windows"))]
 fn get_config_vars(python_path: &Path) -> Result<HashMap<String, String>> {
+    if env::var("CARGO_CFG_TARGET_OS").unwrap() == "windows" {
```

Comment:
What does this refactoring mean?
So `#[cfg(target_os = "~")]` and `env::var("CARGO_CFG_TARGET_OS")` are different?

Reply:
Yes - in `build.rs` I understand that `#[cfg(target_os)]` means the target is the *build script* is being compiled for. This is always the host platform.

`env::var("CARGO_CFG_TARGET_OS")` on the other hand, means the target the crate will be compiled for. So inside the rest of pyo3 source code this will be the value that `#[cfg(target_os)]` matches.